packages/cactus-plugin-persistence-fabric/src/main/typescript/plugin-persistence-fabric.ts

```diff
@@ -0,0 +1,914 @@
+      request: undefined,
+    };
+    this.synchronizationGo = true;
+    do {
+      if (this.lastBlock > this.lastSeenBlock) {
+        try {
+          block = await this.apiClient.getBlockV1({
+            channelName: this.ledgerChannelName,
+            gatewayOptions: this.gatewayOptions,
+            query: {
+              blockNumber,
+            },
+          });
+        } catch (error) {
+          this.log.info("Last block in ledger", tempBlockNumber - 1);
+        }
+
+        if (block.status == 200) {
+          const tempBlockstep1 = JSON.stringify(block.data);
+          const tempBlock = JSON.parse(tempBlockstep1);
+          // Put scrapped block into database
+          this.log.info(tempBlock);
+
+          // if (!this.dbClient.isConnected) {
+          //   await this.dbClient.connect();
+          //   this.isConnected = true;
+          // }
+
+          await this.migrateBlockNrWithTransactions(blockNumber);
+          this.lastSeenBlock = tempBlockNumber;
+          tempBlockNumber = tempBlockNumber + 1;
+          blockNumber = tempBlockNumber.toString();
+        }
+      } else {
+        this.missedBlocks.push(blockNumber.toString());
+        this.howManyBlocksMissing = this.howManyBlocksMissing + 1;
+      }
+      if (this.synchronizationGo && this.lastSeenBlock >= this.lastBlock) {
+        this.synchronizationGo = false;
+      }
+    } while (this.synchronizationGo);
+    return "done";
+  }
+
+  /**  Synchronization of blocks
+   * - Synchronize entire ledger state
+   * with the database as far as the lastBlockInLedger shows ( triggered more than once )
+   * @returns string promise lastBlock number after finishing the process
+   *
+   */
+  // NOTE: this function can loop into very long almost infinite loop or even
+  // infinite loop
+  async continuousBlocksSynchronization(
+    signingCredential: FabricSigningCredential,
+  ): Promise<string> {
+    this.lastSeenBlock = await this.dbClient.getMaxBlockNumber();
+    this.synchronizationGo = true;
+    let tempBlockNumber = this.lastSeenBlock; //last block migrated to database
+    let blockNumber = tempBlockNumber.toString();
+    this.lastBlock = await this.lastBlockInLedger(signingCredential);
+    let block: AxiosResponse<GetBlockResponseV1> = {
+      data: { decodedBlock: {} },
+      status: 0,
+      statusText: "",
+      headers: undefined,
+      config: {},
+      request: undefined,
+    };
+    let checkBlock = true;
+    do {
+      if (this.lastBlock > this.lastSeenBlock) {
+        try {
+          block = await this.apiClient.getBlockV1({
+            channelName: this.ledgerChannelName,
+            gatewayOptions: this.gatewayOptions,
+            query: {
+              blockNumber,
+            },
+          });
+        } catch (error) {
+          this.log.info("Last block in ledger", tempBlockNumber - 1);
+        }
+
+        if (block.status == 200) {
+          const tempBlockstep1 = JSON.stringify(block.data);
```

Comment:
Can we straightaway get the `tempBlock` var by executing JSON.parse(JSON.stringify(block.data))?

Reply:
ok removing